src/components/sequencing/CommandPanel/CommandDictionary.svelte

```diff
@@ -0,0 +1,287 @@
+  import { getTarget } from '../../../utilities/generic';
+  import { showFailureToast, showSuccessToast } from '../../../utilities/toast';
+  import { tooltip } from '../../../utilities/tooltip';
+  import Input from '../../form/Input.svelte';
+  import CommandArgument from './CommandArg.svelte';
+
+  export let commandDictionary: CommandDictionary;
+  export let selectedCommandDefinition: (FswCommand | HwCommand) | null = null;
+
+  type SearchField = 'command_name' | 'enum' | 'argument_name';
+
+  const dispatch = createEventDispatcher<{
+    selectCommandDefinition: (FswCommand | HwCommand) | null;
+  }>();
+
+  const searchFieldMap: Record<SearchField, string> = {
+    argument_name: 'argument names',
+    command_name: 'command names',
+    enum: 'enums',
+  };
+
+  let commandList: (FswCommand | HwCommand)[] = [];
+  let commandResults: (FswCommand | HwCommand)[] = [];
+  let searchField: SearchField = 'command_name';
+  let searchValue: string = '';
+
+  $: commandList = [...commandDictionary.fswCommands, ...commandDictionary.hwCommands];
+  $: commandResults = commandList;
+  $: {
+    switch (searchField) {
+      case 'enum': {
+        commandResults = commandList.filter(command => {
+          return (
+            isFswCommand(command) &&
+            command.arguments.find(argument => {
+              return isFswCommandArgumentEnum(argument) && argument.enum_name.toLowerCase().includes(searchValue);
+            }) !== undefined
+          );
+        });
+        break;
+      }
+      case 'argument_name': {
+        commandResults = commandList.filter(command => {
+          return (
+            isFswCommand(command) &&
+            command.arguments.find(argument => {
+              return argument.name.toLowerCase().includes(searchValue);
+            }) !== undefined
+          );
+        });
+        break;
+      }
+      case 'command_name':
+      default: {
+        commandResults = commandList.filter(command => {
+          return command.stem.toLowerCase().includes(searchValue);
+        });
+        break;
+      }
+    }
+  }
+
+  function onClear() {
+    dispatch('selectCommandDefinition', null);
+  }
+
+  async function onCopy() {
+    if (selectedCommandDefinition) {
+      try {
+        await navigator.clipboard.writeText(`${selectedCommandDefinition.stem}`);
+        showSuccessToast(`"${selectedCommandDefinition.stem}" copied to clipboard`);
+      } catch {
+        showFailureToast(`Error copying "${selectedCommandDefinition.stem}" to clipboard`);
+      }
+    }
+  }
+
+  function onSearch(event: Event) {
+    const { value } = getTarget(event);
+    searchValue = `${value}`.toLowerCase();
+  }
+
+  function onSelectCommandDefinition(commandDefinition: FswCommand | HwCommand) {
+    dispatch('selectCommandDefinition', commandDefinition);
+  }
+</script>
+
+<div class="command-dictionary-container">
+  <div class="dictionary-name">
```

Comment:
Resizes from 35px -> 38px when switching between command dictionary and specific command subview

Reply:
Also noticed that the header below Selected Command is 37px due to the button height being 20px